scheduler/local_scd_server.py

```diff
@@ -0,0 +1,320 @@
+
+    args = parser.parse_args()
+
+    scd_dir = args.scd_dir
+    scd_logs = scd_dir + "/logs"
+
+    emailer = email_utils.Emailer(args.emails)
+
+    if not os.path.exists(scd_dir):
+        with open(scd_logs + "/server_error.txt", 'w') as f:
+            f.write('Supplied local server scd folder does not exist')
+        emailer.email_log("Error starting local SCD server", scd_logs + "/server_error.txt")
+        raise ValueError("Supplied local server scd folder does not exist")
+
+    try:
+        os.makedirs(scd_logs)
+    except FileExistsError:
+        pass
+
+
+    sites = list(EXPERIMENTS.keys())
+    site_scds = [scd_utils.SCDUtils("{}_scd.txt".format(s)) for s in sites]
+    swg = SWG(scd_dir)
+
+    while True:
+        if swg.new_swg_file_available():
+            swg.pull_new_swg_file()
+
+            site_experiments = [swg.parse_swg_to_scd(EXPERIMENTS[s], s) for s in sites]
+
+            errors = False
+            today = datetime.datetime.utcnow()
+            scd_error_log = "/scd_errors.{}{}{}".format(today.year, today.month, today.day)
+            for se, site_scd in zip(site_experiments, site_scds):
+                for ex in se:
+                    try:
+                        site_scd.add_line(ex['mmddyyyy'], ex['hhmm'], ex['experiment'])
+                    except ValueError as e:
+                        error_msg = ("{logtime}: Unable to add line with parameters:\n"
+                                     "\t {date} {time} {experiment}\n"
+                                     "\t Exception thrown:\n"
+                                     "\t\t {exception}\n")
+                        error_msg = error_msg.format(logtime = today.strftime("%c"),
+                                                        date=ex['mmddyyyy'],
+                                                        time=ex['hhmm'],
+                                                        experiment=ex['experiment'],
+                                                        exception=str(e))
+
+                        with open(scd_logs + scd_error_log, 'a') as f:
+                            f.write(error_msg)
+
+                        errors = True
+                    except FileNotFoundError as e:
+                        error_msg = "SCD filename: {} is missing!!!".format(site_scd.scd_filename)
+                        with open(scd_logs + scd_error_log, 'a') as f:
+                            f.write(error_msg)
+
+                        errors = True
+
+
+            subject = "Scheduling report for swg lines"
+            if not errors:
+                success_msg = "All swg lines successfully scheduled.\n"
+                for site, site_scd in zip(sites, site_scds):
+                    next_month = swg.get_next_month()
+                    mmddyyyy = next_month.strftime("%x")
+                    hhmm = next_month.strftime("%H:%M")
+
+                    new_lines = site_scd.get_relevant_lines(mmddyyyy, hhmm)
```

Comment:
instead of using next month, can we just print current and future lines (ie, use today instead of next_month?). I think that's nicer to just include all information in schedule from the current point

Reply:
sure.